src/sensor/ping.cpp

```diff
@@ -359,25 +359,33 @@ void Ping::flash(const QString& fileUrl, bool sendPingGotoBootloader, int baud,
 
     qCDebug(PING_PROTOCOL_PING) << "Finish connection.";
     // TODO: Move thread delay to something more.. correct.
-    QThread::msleep(1000);
-    link()->finishConnection();
+    auto flashSensor = std::function<void(void)>();
+    auto finishConnection = std::function<void(void)>();
+
+    flashSensor = [=] {
```

Comment:
sort of what tomaz said, why don't you just declare + define in one line?

Reply:
Checks my previous answer: 
https://github.com/bluerobotics/ping-viewer/pull/866#discussion_r358879255